src/wildcard.c

```diff
@@ -13,34 +13,46 @@ match_t Wildcard_MatchChar(const char *pattern, size_t p_len, const char *str, s
   const char *ns_itr = NULL;
   int i = 0;
   while (++i) {
-    //printf("%d ", i);
-    if (pattern_end != pattern_itr) {
+    if (pattern_end > pattern_itr) {
       const char c = *pattern_itr;
-      if ((str_end != str_itr) && (c == *str_itr || c == '?')) {
+      if ((str_end > str_itr) && (c == *str_itr || c == '?') && (c != '*')) {
+        // Equal characters or '?' match --> advance both pointers
         ++str_itr;
         ++pattern_itr;
         continue;
       } else if (c == '*') {
```

Comment:
optional.
since you are checking c != '*' at the start maybe the if which checks c == '*' can be at the start?

Reply:
Right 👍